util/progress/printer.go

```diff
@@ -34,15 +41,25 @@ func NewPrinter(ctx context.Context, out console.File, mode string) *Printer {
 		done:   doneCh,
 	}
 
-	if v := os.Getenv("BUILDKIT_PROGRESS"); v != "" && mode == "auto" {
+	if v := os.Getenv("BUILDKIT_PROGRESS"); v != "" && mode == PrinterModeAuto {
 		mode = v
 	}
 
 	go func() {
 		var c console.Console
-		if cons, err := console.ConsoleFromFile(out); err == nil && (mode == "auto" || mode == "tty") {
-			c = cons
+		switch mode {
+		case PrinterModeQuiet:
+			devNull, err := os.Open(os.DevNull)
+			if err == nil {
+				out = devNull
+			}
+			fallthrough
+		case PrinterModeAuto, PrinterModeTty:
+			if cons, err := console.ConsoleFromFile(out); err == nil {
+				c = cons
+			}
 		}
+
```

Comment:
This could be connected to `-q` as well. Otherwise, the ux is confusing unless we want to completely remove `-q`. But there are some things to figure out like outputs can pipe to stdout if image ID is printed there so I guess it could be follow up. Another way to make it faster is to just read the channel here directly until end if the content is disarded anyway. `DisplaySolveStatus` does need to analyze the objects and sort/cache them so has some overhead.